pokedex_async_redux/lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -1,68 +1,141 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 import 'package:pokedex_asyn_redux/api/model/pokemon.dart';
 import 'package:pokedex_asyn_redux/features/pokemon_details/pokemon_details_connector.dart';
 import 'package:pokedex_asyn_redux/features/pokemon_overview/widgets/pokemon_card.dart';
+import 'package:pokedex_asyn_redux/features/pokemon_overview/widgets/pokemon_search_bar.dart';
 import 'package:pokedex_asyn_redux/utils/async.dart';
 import 'package:pokedex_asyn_redux/utils/colors.dart';
+import 'package:pokedex_asyn_redux/utils/number_constants.dart';
 import 'package:pokedex_asyn_redux/utils/string_constants.dart';
 import 'package:pokedex_asyn_redux/widgets/reusable_appbar.dart';
 import 'package:pokedex_asyn_redux/widgets/spacing.dart';
 
-class PokemonOverviewPage extends StatelessWidget {
+class PokemonOverviewPage extends StatefulWidget {
   const PokemonOverviewPage({
     required this.pokemons,
+    required this.searchedPokemons,
+    required this.filterPokemons,
+    required this.clearSearchedPokemons,
     Key? key,
   }) : super(key: key);
 
   final Async<List<Pokemon>> pokemons;
+  final List<Pokemon> searchedPokemons;
+  final VoidCallback clearSearchedPokemons;
```

Comment:
kindly rename this to `onClearSearchedPokemons`, same goes with the functions declared in your vm 🙂